lib/std/build.zig

```diff
@@ -2271,6 +2283,18 @@ pub const LibExeObjStep = struct {
         }
 
         if (self.target.isDarwin()) {
+            if (self.syslibroot) |path| {
+                try zig_args.append("-syslibroot");
+                try zig_args.append(path);
+            } else {
+                if (self.target.isNative()) {
+                    const syslibroot = try std.zig.system.getSDKPath(builder.allocator);
```

Comment:
We shouldn't need this call in std.build because we expect this to work:

```
zig build-exe something_that_requires_libsysroot.zig
```

This means if the target OS is native and we omit the -syslibroot parameter then the compiler will figure it out (probably by doing getSDKPath itself).